### PR TITLE
Unique hostnames via host shortname

### PR DESCRIPTION
**Pull request by ehelms**
This won't make sense on a single, isolated libvirt setup. But for some larger hypervisor environments, or if using a Foreman/Katello to register to for content, having the same hostname used by multiple users can result in missing data.

### CODE REVIEW

**Comment by ehelms** [comment]
@ekohl or @evgeni thoughts?

**Comment by evgeni** [comment]
Yes! I wanted to add something like this last week already, but then forgot!

**Comment by ekohl** [comment]
Interesting thought. I was playing with the idea of making the whole domain configurable so maybe the variable should be ```domain```. Then at some point we should add a config option somewhere.

Another thing I'm wondering: how much of this breaks assumptions that the domain is ```example.com``` in our playbooks?

**Comment by ehelms** [comment]
I can't think of anything that assumes `example.com` @ekohl . I can add a domain parameter to make it configurable, but I feel like in the base case this is still a viable option to achieve automatic uniqueness.

File: lib/forklift/box_distributor.rb [hunk 1]
```diff
@@ -67,10 +67,11 @@ def define_vm(config, box = {})
 
         machine.vm.box_url = box.fetch('box_url') if box.key?('box_url')
 
+        host_short_name = `hostname -s`.strip
```
Comment (ekohl) [inline lib/forklift/box_distributor.rb]: I'd prefer ```domain = "#{`hostname -s`.strip}.example.com"```. The configurability can be added later.

**Comment by evgeni** [comment]
We could use `etc_hosts_domain` from `roles/etc_hosts/vars/main.yml`? As that defines the current `example.com` base for a lot of places.

**Comment by ehelms** [comment]
@evgeni use it how?

**Comment by evgeni** [comment]
Argh, ewoud confused me :D

@ekohl suggested to use `{{ ansible_domain }}`, and my braint went ahead and thought "we could also do `{{ etc_hosts_domain | default(ansible_domain) }}` thus alowing the user to override it easily".

But we're in forklift code here, none of the ansible vars are available yet.

**Comment by ehelms** [comment]
Oi, I take that back. Quite a few things are married to `example.com` such as provisioning setup and etc_hosts.

**Comment by ehelms** [comment]
Updated the playbooks and roles.

File: roles/freeipa_server/defaults/main.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ katello_server_ip: "{{hostvars[katello_server_origin]['ansible_ssh_host']}}"
 
 freeipa_server_ip: "{{ ansible_eth0.ipv4.address }}"
 freeipa_server_realm: "EXAMPLE.COM"
```
Comment (ekohl) [inline roles/freeipa_server/defaults/main.yml]: I don't know FreeIPA but should this be ```{{ ansible_domain|upper }}```?

File: pipelines/pipeline_smartproxy_powerdns.yml [hunk 3]
```diff
@@ -31,7 +31,7 @@
     powerdns_db_pass: "ajooy7HeeCh8loo"
     powerdns_conf_mode: "0644"
     powerdns_zones:
-      - "example.com"
+      - "{{ ansible_domain }}"
```
Comment (ekohl) [inline pipelines/pipeline_smartproxy_powerdns.yml]: This breaks the bats tests since ```fb-proxy-dns.bats``` hardcodes the domain to ```example.com```.

File: roles/pulp_repositories/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+---
```
Comment (ekohl) [inline roles/pulp_repositories/defaults/main.yml]: Unrelated change that snuck in with ```git add .````?

File: roles/freeipa_server/defaults/main.yml (hunk 2 above)
Comment (ehelms) [inline roles/freeipa_server/defaults/main.yml]: Good find, my grepping failed.

File: pipelines/pipeline_smartproxy_powerdns.yml (hunk 3 above)
Comment (ehelms) [inline pipelines/pipeline_smartproxy_powerdns.yml]: Hrmm, so should I revert this or update the bats tests to grab the domain?

Comment (ekohl) [inline pipelines/pipeline_smartproxy_powerdns.yml]: For now I'd revert it since the actual domain doesn't really matter

**Comment by ehelms** [comment]
Fully updated

**Comment by evgeni** [comment]
LGTM!
